.semaphore/docker-hub.yml

```diff
@@ -12,8 +12,10 @@ blocks:
       prologue:
         commands:
           - checkout --use-cache
-          - export DOCKER_IMAGE_TAG=${SEMAPHORE_GIT_SHA:0:7}
-          - cache restore ${DOCKER_IMAGE_TAG}
+          - export DOCKER_IMAGE_TAG=${SEMAPHORE_GIT_TAG_NAME:-${SEMAPHORE_GIT_SHA}}
```

Comment:
we should use the short SHA version herre `${SEMAPHORE_GIT_SHA:0:7}`

Reply:
it's amibigious in very rare cases... and pushing a sha is bad anyway....

Reply:
why is pushing an sha a bad idea? using the tag commits sha hash is a good choice imho. It's an expressive and immutable tag that we can identify across git, docker hub, etc.

using a branch or a more generic name ( even latest ) is a bad practice because it's not expressive.

Reply:
there is no use case for using a git sha...

Reply:
it's just polluting the tags section on docker hub...

Reply:
the whole point of a tag is to point to have a single unique and immutable reference to the id of a docker image. The ID of a docker image is nothing else than a SHA256 hash of the images configuration file.

We can agree on whatever scheme to use for tagging images but we should NOT use the same tag for multiple builds! Each build should be a unique version. Of course, that has nothing to do with the fact that we don't need to push every version to hub.docker.com. Pushing releases that follow semver only is fine.

Reply:
so if we don't use git SHA, we should something else that is unique, expressive and immutable. Git SHA makes sense to me since it points a specific version, similar to a Git Tag - which also points to a specific commit hash.